### DISCUSSION TITLE
在BaseNode或BaseEdge和IBaseModel中有办法获取到LogicFlow实例吗

### DISCUSSION THREAD

**Discussion by cpf23333** [didi/LogicFlow#1287] [Q&A]
如题，目前节点使用的一些数据需要从LogicFlow实例上获取
除了通过一个创建函数传入LogicFlow实例，返回节点类定义外，还有别的办法吗

**Comment by boyongjiong** [comment]
能举例说一下哪些数据需要从 LogicFlow 的实例上获取吗？是现有的数据还是有一些你自定义的数据呢？

**Reply by cpf23333** [reply]
我拓展了LogicFlow类，在上面加了些属性和方法，节点的model和view类需要根据我在LogicFlow实例上存储的数据来展示不同的内容和执行不同的逻辑

在你们的文档项目
https://github.com/Logic-Flow/docs/blob/b861c7945911b74c30ce9323e40ab5083d5c19dc/demo/bpmn-vue/src/plugin/logicflow/elements/Event/index.ts#L40
中也有类似的需求

**Reply by boyongjiong** [reply]
<img width="852" alt="image" src="https://github.com/didi/LogicFlow/assets/8553969/3d0330ad-4cef-4568-bc8d-ddd181510144">

你看下源码中 register 上面的注释，这个方式二是否符合你的预期，试一下呢？

**Reply by boyongjiong** [reply]
LogicFlow.tsx -> register 函数

**Reply by cpf23333** [reply]
不符合，要定义的节点可能很多，对节点的定义的代码也可能很长，不可能写到类实例化之后的地方

但如果拆分到不同文件，就需要变成我提出的问题那里的解决方案了

因为第二种定义，函数的参数中并没有提供LogicFlow实例，只是因为类定义在lf声明的代码块下，所以可以访问到lf而已
